tests/sqlparser_common.rs

```diff
@@ -1594,6 +1594,15 @@ fn parse_limit_accepts_all() {
     );
 }
 
+#[test]
+fn parse_json_using_colon() {
+    one_statement_parses_to("SELECT field:key FROM t", "SELECT field ->> key FROM t");
```

Comment:
@yuval-illumex would it be possible to create an expression to maintain the original syntax?

This transformation seems to be using semantic analysis (converting `:` to `->>`) which, even though could be considered right depending on the dialect, doesn't maintain the original expression syntax.